### ISSUE TITLE
Power surges - extra complication for power management.

### ISSUE THREAD

**Issue by Screemonster** [SpadesNeil/VOREstation#655]
At the moment, there's no actual consequences for engineers deciding to just say "fuck it", hardwire both the engine and solars, and call it a day, besides the effect it has on shock damage (and since damage to a single limb is capped at 75, people can't be instakilled by a shock anyway)

A lil' discussion in skype/discord suggested adding some _fun_ to the power system if people decide to just dump more power into the grid than is needed.

Rather than doing anything based on the power in the wires (which would cause eternal problems in the engine room unless some sort of heavy power conduit was implemented) I was thinking of putting something into the APC code based upon the power being fed into them - nothing would happen below a safe threshold, with more "interesting" effects taking place as the wattage increases.

For reference, 200kW is more than enough for any individual department, and the roundstart SMES setting (without any substations in use) is 500kW. The APCs in maintenance and the solars are hooked directly to the main distribution grid and cannot be "stepped down" via a substation, so they might wind up set to ~750kW if the substations are all being charged at once.

Ideally, below 500k, only minor problems should trigger, and then only rarely. Below 200k there should be no effects at all.

Tweaking the probability curve of the frequency and severity of the events is a maths thingy on its own - maybe something like a "pick a number between 0 and the current power level, then look up a list of results" so more severe effects are outright excluded below a certain wattage.
If the APCs breaker is off, then no effects should occur.

Possible effects include:
Pick a random light and blow it. This is probably the worst that should happen below 500kW, and shouldn't happen if lights are off.

Above 500k:
Set off some random sparks from a piece of equipment in the area unless equipment power is off.
Blow _all_ the lights in the area unless lighting power is off. (same as the AI's overload ability)

Above 800k: (any less and these will happen in maintenance too often while the substations are charging)
Start a fire in the area (with sparks!)
Trip the main breaker on the APC. (hey, at least nothing else will break!)
Break a piece of equipment unless equipment power is off. (might be tricky as not all equipment has code for this)

Above 1MW:
The cell in the APC explodes.

Above this, as the power rises it'll just make the more servere effects more likely.

Basically, add some "fun" to power engineering. :3

**Comment by Screemonster** [comment]
Oh, and "trip the breaker" should probably be the most likely outcome of the >800k and >1M lists, just to stop griefers being able to max the power and explode everything forever.

**Comment by antsnap** [comment]
I'm against, although the concept is neat.

My main concern is that the job for Engi's on most servers is.
1. Set up power
2. Fix stuff antags/meteors/events break
3. Optimize/play-with the engines
On most servers (2.) happens a lot. We are different (not often antags), so that might lead to the desire to make power break as a 'fun' RNG. The issue is, this isn't just RNG, this is inevitable. And, because of small power changes triggering procs, even if made probability based, it will happen likely after each SMES tweak (the chance would be APC based right? So, with N, each proc you have N * proc_chance, chance. You make the chance small but...). This would still be ok, _but,_ unlike the other things that could randomly break (lights), this interferes with 3.

You'll notice that every (almost) every engine on servers is 
1. Simple to set-up 
2. Has complex potential for player improvement/fun.

For the singularity engine, an average player might just fill the plasma tanks up and stick them in the collector. However, an inquisitive engineer will want to improve the output by filling the tanks instead with _super cooled plasma_, which allows for more plasma and thus more power from the radiation collectors. The singularity often gets a rep for being "too-easy", but its power output compared to all alternatives but solars is inferior--without cooled-gas-tanks.

The SME can be set up by just toggling the pre-installed valves (and occasionally replacing the nitrogen canisters hooked to it), _but_ the room and engine has a lot of potential for fun hacks. Like the common space-ultra-cooled nitrogen you can work with atmos to work out. An adventurous engineer could also used super cooled plasma as the coolant. You could even filter the heated plasma from the engine for other uses.

Even the more niche engines have advantages, challenges, and balances associated (anti-matter engines generate more power with more cores, but need more fuel).

So, if we implement power-surges, there is now **danger** in doing the above-and-beyond stuff some engineers do. Why would I improve the SME if people will just yell at me cause their lab's APC keeps flicking off? I wouldn't. Also, since I _can't_ wire directly into the grid anymore, I have to use the SMES, which means my power is capped anyway, **so** why would I want to maximize it anymore?

Also unlike other servers, after doing their job to set up power, many engineers run off to engage in scenes. If I'm busy eating micemen in dorms, I probably don't want to get PDA spammed to go diddle the SMES/engine/whatever because RD needs more power/apc-exploded.

But there's usually more than one engineer right? Well yes, at the more peak times, but most of the crew flocks towards medbay and service; so even if there are 3 engineers, 3 hours in, they are either sceneing or SSD.

Power surges are realistic, sure, but so are surge protectors. Immersion wise I consider the APCs also having the function of intelligent and advanced surge protecting. This is somewhat counteracted by the nature of how the airlocks work, but that too can be rationalized as a built in defense system, that contacts the APC upon tampering, to draw what power it can for the shock.

TL;DR I don't consider it more immersive than before. The fun it adds is outweighed by removing fun from engine tweaking. Although it does balance out the server by making engi's do **more** things after initial setup, I don't consider that very good for an ERP server in the way it's described.

**Comment by Screemonster** [comment]
You can still improve the SME, and build whatever wacky-crazy engine you like - SMESs are constructable now so even if you want to build a completely new engine somewhere attached to the station far from the engine room, just build a SMES as a buffer so your megafucktons of power don't get to the main grid.

Or set up the substations. There actually aren't that many APCs connected directly to the distro grid (the red wires) directly. We're talking a few in maintenance and that's it.

That's why I suggested having nothing _serious_ happen until the power reached 800k at least. A small chance of some sparks (which really just add some atmosphere unless someone's flooded plasma into the tunnels) or the lights blowing (and maintenance is pretty dark in most places anyway)

Shield generators need to be built on a wire and don't use the APC for the area - just find a red cable, park the capacitor on a node and turn it on, and you don't need to have the APC in the area run on any more power power than normal. (the new map is good about that - cables from engineering to the substation inputs are red, while those from the substations to the APCs are green)
